ddtrace/internal/telemetry/writer.py

```diff
@@ -0,0 +1,209 @@
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ...internal import forksafe
+from ...settings import _config as config
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..compat import monotonic
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..runtime import get_runtime_id
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .data import get_application
+from .data import get_host_info
+
+
+log = get_logger(__name__)
+
+
+def _get_interval_or_default():
+    return float(get_env("instrumentation_telemetry", "interval_seconds", default=60))
+
+
+AGENT_URL = get_trace_url()
+ENDPOINT = "telemetry/proxy/api/v2/apmtelemetry"
```

Comment:
TODO: Move `ENDPOINT` to be a class constant on `TelemetryWriter`
TODO: Add a class property for the agent url and call `get_trace_url()` in the `__init__` of `TelemetryWriter`